lab_2_retrieval_w_bm25/main.py

```diff
@@ -120,7 +187,26 @@ def calculate_bm25(
 
     In case of corrupt input arguments, None is returned.
     """
-
+    if (not isinstance(vocab, list) or not all(isinstance(word, str) for word in vocab)
+            or not isinstance(document, list) or not
+            all(isinstance(word, str) for word in document) or not isinstance(k1, float)):
+        return None
+    if (not isinstance(b, float) or not isinstance(avg_doc_len, float)
+    or not isinstance(doc_len, int) or not isinstance(idf_document, dict)
+    or isinstance(doc_len, bool)):
+        return None
+    if not all((isinstance(key, str) and isinstance(value, float) for key, value in
+                idf_document.items())):
+        return None
+
+    bm25 = {}
+    for word in set(idf_document) | set(document):
+        if word not in idf_document:
```

Comment:
could it be that we are looping through every `word` in `idf_document` and we don’t have `word` in `idf_document`?

Reply:
I think I'm actually really confused about what to put after "for word in", since no matter what i put in there, the result doesn't change..